### ISSUE TITLE
Cannot create image collections from earthdata cubes, despite setting gdal_http_headers token

### ISSUE THREAD

**Issue by cboettig** [appelmar/gdalcubes#90]
As you know, NASA's EarthData program is a major provider of geospatial datasets we might want to access with gdalcubes.  Normally this works ok, but I'm having trouble when I need to directly create a cube for data where there is no STAC catalog.

A reprex is worth a thousand words:

```r

## Small helper package to create a token for NASA EarthData

# remotes::install_github("boettiger-lab/earthdatalogin")
library(earthdatalogin)
header = edl_set_token(format="header", set_env_var = FALSE)
Sys.setenv("GDAL_HTTP_HEADERS" = header)

## 4 example URLs
urls <- c("/vsicurl/https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MUR-JPL-L4-GLOB-v4.1/20200101090000-JPL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc",
"/vsicurl/https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MUR-JPL-L4-GLOB-v4.1/20200102090000-JPL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc",
"/vsicurl/https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MUR-JPL-L4-GLOB-v4.1/20200103090000-JPL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc",
"/vsicurl/https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MUR-JPL-L4-GLOB-v4.1/20200104090000-JPL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc")

## Stars has no problem accessing these URLs after we have set the env var above
library(stars)
r <- read_stars(urls)

## gdalcubes fails
library(gdalcubes)
gdalcubes_set_gdal_config("GDAL_HTTP_HEADERS", header)

dates <- stringr::str_extract(basename(urls), "\\d{14}") |> lubridate::as_datetime()
col <- create_image_collection(urls, date_time = dates )
```

Note that `stars` has no problem taking advantage of the `GDAL_HTTP_HEADERS` bearing the auth token, but gdalcubes fails to respect this header (requires GDAL >= 3.6.1, I'm using 3.7 anyway)

but create_image_collection skips all 4 files: 

```
WARNING] GDAL failed to fetch geotransform parameters for '/vsicurl/https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MUR-JPL-L4-GLOB-v4.1/20200101090000-JPL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc'; dataset will be skipped
[WARNING] GDAL failed to fetch geotransform parameters for '/vsicurl/https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MUR-JPL-L4-GLOB-v4.1/20200102090000-JPL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc'; dataset will be skipped
[WARNING] GDAL failed to fetch geotransform parameters for '/vsicurl/https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MUR-JPL-L4-GLOB-v4.1/20200103090000-JPL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc'; dataset will be skipped
[WARNING] GDAL failed to fetch geotransform parameters for '/vsicurl/https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MUR-JPL-L4-GLOB-v4.1/20200104090000-JPL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc'; dataset will be skipped
```

Looks like `create_image_collection()` might be calling gdal in a way that doesn't respect the `gdalcubes_set_gdal_config` or the system env var?

**Comment by cboettig** [comment]
ha! I see my mistake -- gdalcubes _is_ getting the data, but these ncdf files lack a CRS that it can identify, so it's choosing to skip them.  Is there a simple way I can tell it to assume these are WGS 84?

**Comment by cboettig** [comment]
ok, we can pass the CRS with the appropriate gdal prefix and suffix to define format, band, extent, and crs:

```
prefix <-  "vrt://NETCDF:"
suffix <- ":analysed_sst?a_srs=OGC:CRS84&a_ullr=-180,90,180,-90"
uri <- paste0(prefix, urls, suffix)


library(gdalcubes)
gdalcubes_set_gdal_config("GDAL_HTTP_HEADERS", header)

dates <- stringr::str_extract(basename(urls), "\\d{14}") |> lubridate::as_datetime()
col <- create_image_collection(uri, date_time = dates )
```

Thanks @mdsumner for this solution and apologies for the noise.

**Comment by mdsumner** [comment]
just note this is only (vrt://) syntax args available since GDAL 3.7.0, you can create full VRT in previous versions but it has to open the dataset, assign the metadata fixes, and rewrite to actual VRT file

(also btw we're planning on uploading the *full* set of GHRSST time series as  COGs to source.coop, within a week or two - but we'll probably use your  gdal headers trick rather than the set of hacks I was considering!)

**Comment by mdsumner** [comment]
also fwiw, I've decided that the correct a_ullr is -180,89.9905,180,-89.9905 given the weird values in the netcdf and the 36000x17999 dim - with -90,90 you really need 18000 rows, turns out xarray does something odd with adding an extra row/col (assuming area vs centre interpretation presumably), and we're still exploring that
